Sources_v3/StreamChat/WebSocketClient/WebSocketClient.swift

```diff
@@ -134,27 +131,27 @@ class WebSocketClient {
     ///
     /// Calling this method has no effect is the web socket is already connected, or is in the connecting phase.
     func connect() {
+        guard let endpoint = connectEndpoint else {
+            log.assertationFailure("Attempt to connect `web-socket` while endpoint is missing")
+            return
+        }
+
         switch connectionState {
         // Calling connect in the following states has no effect
         case .connecting, .waitingForConnectionId, .connected(connectionId: _):
             return
         default: break
         }
         
-        // Engine needs to be recreated if the connection endpoint has changed. For example, when a new user is
-        // logged in, or when an auth token is refreshed.
-        if engineNeedsToBeRecreated {
-            engineNeedsToBeRecreated = false
-            engine = createEngine()
-        }
+        engine = createOrReuseEngine(for: endpoint)
         
         // Cancel the reconnection timer if exists
         reconnectionTimer?.cancel()
         
         connectionState = .connecting
         
         engineQueue.async { [engine] in
-            engine.connect()
+            engine?.connect()
```

Comment:
This looks like a potential silent failure point to me. Either `!` or unwrap properly and log an error if `engine` is `nil`. It's probably OK for `disconnect` and `sendPing` but here in `connect` we should treat it differently.

Reply:
fixed in c732d658ca9625bd74d35c32c4d80b3a70871a0e. Used force uwrap cause after `createEngineIfNeeded` is call the engine should never become `nil`